server/src/com/cloud/network/NetworkServiceImpl.java

```diff
@@ -2111,8 +2098,9 @@ public Network updateGuestNetwork(final long networkId, String name, String disp
             }
         }
 
-        final Map<String, String> newSvcProviders = networkOfferingChanged ? _networkMgr.finalizeServicesAndProvidersForNetwork(
-                _entityMgr.findById(NetworkOffering.class, networkOfferingId), network.getPhysicalNetworkId()) : new HashMap<String, String>();
+        final Map<String, String> newSvcProviders = networkOfferingChanged
+                ? _networkMgr.finalizeServicesAndProvidersForNetwork(_entityMgr.findById(NetworkOffering.class, networkOfferingId), network.getPhysicalNetworkId())
+                : new HashMap<String, String>();
```

Comment:
Recognizing it is out of scope of this patch, but would it be possible to change ``new HashMap<String, String>()`` to ``Collections.emptyMap()`` to avoid unnecessary object creation when ``networkOfferingChanged`` is ``false``.